extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -0,0 +1,221 @@
+		return (await this.getValue())! <= ((await this.getVariableValue(this.target))!);
+	}
+}
+
+export class GreaterThanOrEqualsValidation extends Comparison {
+	async isComparisonSuccessful() {
+		return (await this.getValue())! >= ((await this.getVariableValue(this.target))!);
+	}
+}
+
+
+/**
+ *	removes validation message corresponding to this validator from the @see dialogMessage string.
+ *
+ * @param dialogMessage
+ * @param message
+*/
+export function removeValidationMessage(dialogMessage: azdata.window.DialogMessage, message: string): azdata.window.DialogMessage {
+	if (dialogMessage === undefined) {
+		return dialogMessage;
+	}
+	if (dialogMessage.description) {
+		return {
+			text: dialogMessage.text,
+			level: dialogMessage.level,
+			description: getStrippedMessage(dialogMessage.description, message)
+		};
+	} else {
+		return {
+			text: getStrippedMessage(dialogMessage.text, message),
+			level: dialogMessage.level,
+			description: dialogMessage.description
+		};
+	}
+}
+
+function getStrippedMessage(originalMessage: string, message: string) {
+	if (originalMessage.includes(message)) {
+		const messageWithLineBreak = message + EOL;
+		const searchText = originalMessage.includes(messageWithLineBreak) ? messageWithLineBreak : message;
+		const newMessage = originalMessage.replace(searchText, '');
+		return newMessage;
+	} else {
+		return originalMessage;
+	}
+}
+
+export function createValidation(validation: ValidationInfo, valueGetter: ValueGetter, variableValueGetter?: VariableValueGetter): Validation {
+	switch (validation.type) {
+		case ValidationType.Regex: return new RegexValidation(<RegexValidationInfo>validation, valueGetter);
+		case ValidationType.IsInteger: return new IntegerValidation(<IntegerValidationInfo>validation, valueGetter);
+		case ValidationType.LessThanOrEqualsTo: return new LessThanOrEqualsValidation(<ComparisonValidationInfo>validation, valueGetter, variableValueGetter!);
+		case ValidationType.GreaterThanOrEqualsTo: return new GreaterThanOrEqualsValidation(<ComparisonValidationInfo>validation, valueGetter, variableValueGetter!);
+		default: throw new Error(`unknown validation type:${validation.type}`); //dev error
+	}
+}
+
+export async function validateAndUpdateValidationMessages(component: InputComponent, container: azdata.window.Dialog | azdata.window.Wizard, validations: Validation[] = []): Promise<ValidationResult> {
+	let dialogMessage = container.message;
+	const validationStates = await Promise.all(validations.map(validation => validation.validate())); // strip off validation messages corresponding to successful validations
+	validationStates.filter(state => state.valid).forEach(v => dialogMessage = removeValidationMessage(dialogMessage, v.message!));
+	const failedStates = validationStates.filter(state => !state.valid);
+	if (failedStates.length > 0) {
+		container.message = getDialogMessage([dialogMessage?.description ?? dialogMessage?.text, failedStates[0].message!]);
```

Comment:
Yes because DialogMessage is shared across multiple fields, each of which could be posting/removing messages from this dialogMessage based on which validations have succeeded/failed.

Reply:
I didn't look - so is getDialogMessage called in multiple places then?

We're doing all the validations here though - why wouldn't we display all the messages of failures instead of just one? That's just going to be annoying to try and fix them all if there's multiple I would think.

Reply:
Also, a single component can emit multiple validation failures.

Reply:
if you read that closely you will see that we pass in an array of size 2. The first element is the string that was already in the dialogMessage and 2nd element is the message corresponding to the current failed validation that we are processing.

Reply:
Ah, I missed that.

I don't understand the "current failed validation we are processing". You went through all the validations in the lines above and are processing the results here. You could have multiple validations fail each with their own message - why are you not including those?